site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/support/components/SupportForm.js

```diff
@@ -120,12 +125,32 @@ class SupportForm extends Component {
             >
                 {({ handleSubmit, setFieldValue, values }) => {
                     let dropzoneParams = {
+                        validator: (file) => {
+                            const { totalFileSize } = this.state;
+                            if(totalFileSize + file.size > maxFileSize) {
+                                return {
+                                    code: "file-too-large",
+                                    message: ""
+                                }
+                            }
+                            this.setState({totalFileSize: totalFileSize + file.size});
+                        },
+                        onDrop: () => {
+                            this.setState({ 
+                                fileErrorMessage: "",
+                                totalFileSize: 0
+                            });
+                        },
                         onDropAccepted: (acceptedFiles) => {
                             setFieldValue("files", acceptedFiles);
                         },
                         onDropRejected: (rejectedFiles) => {
-                            // TODO: show error message when files are rejected e.g size limit
-                            console.error(rejectedFiles[0].errors);
+                            const error = rejectedFiles[0].errors[0];
+                            let message = error.message;
+                            if(error.code === "file-too-large") {
+                                message = "Max attachment size exceeded. Please add URLs to the files or make a smaller selection.";
+                            }
```

Comment:
I think it would be better to have this separately defined in an object:
```
ErrorCodes = { 
"file-too-large": "Max attachent ...."
}
```
and then you simply use as
```
this.setState({ fileErrorMessage: ErrorCodes[code] });
```

Reply:
Thanks!!